app/Actions/Albums/PositionData.php

```diff
@@ -2,37 +2,37 @@
 
 namespace App\Actions\Albums;
 
-use App\Actions\Album\Extensions\LocationData;
-use App\Actions\Albums\Extensions\PublicIds;
+use App\Actions\PhotoAuthorisationProvider;
 use App\Models\Configs;
 use App\Models\Photo;
 
 class PositionData
 {
-	use LocationData;
+	protected PhotoAuthorisationProvider $photoAuthorisationProvider;
+
+	public function __construct()
+	{
+		$this->photoAuthorisationProvider = resolve(PhotoAuthorisationProvider::class);
+		// caching to avoid further request
+		Configs::get();
+	}
 
 	/**
 	 * Given a list of albums, generate an array to be returned.
 	 *
-	 * @param Collection[Album] $albums
-	 *
 	 * @return array
 	 */
-	public function do()
+	public function do(): array
```

Comment:
Wouldn't it be better to return a Collection and have the Array conversion later ? *wondering*